### DISCUSSION TITLE
GenerateDepictionMatching2DStructure ignore unmatched molecules

### DISCUSSION THREAD

**Discussion by kathputu** [rdkit/rdkit#5862] [Q&A] [answered]
Hi,

I am running a list of molecules to GenerateDepictionMatching2DStructure. However, occasionally there are some molecules with slightly different substituents and the job stops and returns error message "ValueError: Depict error: Substructure match with reference not found." 

I am wondering if there is a way that the function ignores molecules that doesn't match the reference and continues the job for the rest of the molecules? (They just leave the unmatched molecules as it is.)

Hope this question makes sense. Thank you!

**Comment by ptosco** [comment, marked as answer]
Check the documentation,
https://www.rdkit.org/docs/source/rdkit.Chem.rdDepictor.html#rdkit.Chem.rdDepictor.GenerateDepictionMatching2DStructure
There’s an `acceptFailure` parameter that you may set to `True`.
